spec/system/onboardings/user_completes_onboarding_spec.rb

```diff
@@ -28,26 +30,19 @@
   end
 
   context "when the user has seen onboarding" do
-    before do
-      user.update(saw_onboarding: true)
+    let(:user) { create(:user, password: password, password_confirmation: password) }
```

Comment:
I think it might be a good idea to include `saw_onboarding: true` here instead of relying on the default. That way if the default changes this spec doesn't surprisingly fail.

Reply:
@mstruve Hm, I actually think we should explicitly _not_ have to include `saw_onboarding: true` here. In fact, in https://github.com/forem/forem/pull/11259, I explicitly removed all the instances in our specs where we were [explicitly setting `saw_onboarding` to `true`](https://github.com/forem/forem/pull/11259#discussion_r517022340) 

The reason behind this is that I think that if someone goes into the user factory and _does_ change the default from `true` to `false`, I would actually expect a bunch of specs to start failing. I don't think it would be a surprise because if you change a value in the factory and it is implicitly used by every factory instance, then you'd _want_ the specs to tell you (by failing) about all the places where you were relying on that (plus, a ton of other tests would fail too, I think).

IMO, we should be relying on our tests to tell us when these things change, rather than having to explicitly override a factory attribute for clarity's sake. I personally think it's more confusing when factory instances override a default value with the same value as the default _itself_. If we're overriding a value, I would expect it to be different from the default (otherwise, I kind of feel like the default isn't doing much at that point).

Let me know if that ^^ makes sense or if you totally disagree :)

Reply:
> The reason behind this is that I think that if someone goes into the user factory and does change the default from true to false, I would actually expect a bunch of specs to start failing. 

Ideally, I think specs should only fail bc code is broken. The point of a test, as I see it, is to test the code and alert you to when the code breaks and no longer performs the expected behavior. In this case, the code is not broken we only changed the way we set up the test. The fact that a change in a factory setting breaks a bunch of tests simply bc the test set up changed seems less than ideal and makes it a pain if we want to tweak our factory or other general test settings

Reply:
Chatted with @mstruve about this a bit synchronous, and I think I have a better sense of the issue that she's trying to avoid here.

To close the loop for anyone else who stumbles upon this...we think that [these changes](https://github.com/forem/forem/pull/11259/commits/6b3bef932df71de73fcf7c39491131248239e8ad) should remain, because the value of that attribute isn't pertinent/relevant to those tests. But in this case, it's the fact that I am testing something that _relies_ on the default that could lead to changing the default value of `saw_onbaording` in the user factory, but then being forced to sift through and debug whether the test is failing because of the default value or because of something else. So I think it's valid for us to be explicit here 👍